### ISSUE TITLE
(FreeBSD) Poco Websocket crashing in sendFrame API after applying a “deny” ipfw rule

### ISSUE THREAD

**Issue by sharmavijay1991** [pocoproject/poco#2631]
I am having client and server based processes running.
Client is using Poco library to communicate and receive data from server.

In the case of no data, client and server are exchanging Ping-Pong messages.

**Problem statement:** I have CLIENT-1 a module(written in C++/Poco) and SERVER-1 is server.
1. Initially everything is working fine and they are communicating.
2. SERVER-1 is sending ping messages and CLIENT-1 is responding by pong message.
3. Now if I create a "ipfw" rule to block outgoing data from CLIENT-1 device to SERVER-1, then the API "sendFrame" will crash while sending a pong response.

**I tried:**

a) exception handling but there is no catchable-exception by "catch(std::exception)" and "catch(...)"

b) tried poll command with different different flag, but socket is returning proper state.

c) other ways to find socket state.

Nothing helped here.

**Please help me to resolve this issue OR let me know if this is restriction from POCO side.**

**Other Information:**
Poco Version: 1.9
OS: FreeBSD 10.4
Language: C++

**_Sample-Code:_**

    URI websocket_uri_object(ws_url);

    HTTPSClientSession websocket_session(websocket_uri_object.getHost(), websocket_uri_object.getPort()); 
    websocket_session.setKeepAlive(true);

    Poco::Net::HTTPRequest ws_request(Poco::Net::HTTPRequest::HTTP_GET, websocket_uri_object.getPath(),Poco::Net::HTTPMessage::HTTP_1_1);

    HTTPBasicCredentials cred(username, password); 
    cred.authenticate(ws_request);
    ws_request.setKeepAlive(true); 
    Poco::Net:: HTTPResponse ws_response;
    Poco::Net::WebSocket *ws = new Poco::Net::WebSocket(websocket_session, ws_request, ws_response);

    //--- sending data ---
    //-- some ping/ pong ---
    //--- IPFW rule to block outgoing packets ----
`// Sample rule: ipfw add 00010 deny ip from CLIENT-1_IP to SERVER-1_IP`
    
    received_data_size = ws->receiveFrame(subscriber_update_buffer, flags);  //---- > getting ping.
    ws->sendFrame(pong_data_ptr, pong_size, PONG_FLAGS); // --- > sending pong with blank data of size 0 --> program crashing here.

**Comment by Yuanli-Liu** [comment]
hello, have this issue been solved. I met the same problem as yours.

**Comment by matejk** [comment]
I tried to reproduce this issue on Linux and macOS since I do not have access to a FreeBSD computer. I did not manage to reproduce the reported problem on a pre 1.14 development branch.

When firewall was enabled (Ubuntu ufw), client's requests timed out as they should.

Sample client and server used to reproduce are available on branch [2631_websocket-client-server-firewall](https://github.com/pocoproject/poco/tree/2631_websocket-client-server-firewall).

@sharmavijay1991 , @Yuanli-Liu , @aleks-f .
